apps/bfd-data-npi/src/test/java/gov/cms/bfd/data/npi/lookup/NPIOrgLookupE2E.java

```diff
@@ -0,0 +1,46 @@
+package gov.cms.bfd.data.npi.lookup;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.io.IOException;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/** NPI lookup end to end test. */
+public class NPIOrgLookupE2E {
+
+  /** Global variable for NPILookup. */
+  NPIOrgLookup npiOrgDataLookup;
+
+  /** Global variable for npiOrgDisplay. */
+  Optional<String> npiOrgDisplay;
+
+  /** Global variable for valid npiOrgNumber. */
+  private final String npiOrgNumber = "1992903249";
+
+  /** Global variable for invalidNpiOrgNumber. */
+  private final String invalidNpiOrgNumber = "000";
+
+  /** Setup method. */
+  @BeforeEach
+  void setup() throws IOException {
+    npiOrgDataLookup = NPIOrgLookup.createNpiOrgLookupForProduction();
+    npiOrgDisplay = Optional.empty();
+  }
+
+  /** End to End test for Npi Org Data with real npi number. */
+  @Test
+  public void shouldReturnRealOrgNPIDataReturnsRightOrganization() throws IOException {
+    npiOrgDisplay = npiOrgDataLookup.retrieveNPIOrgDisplay(Optional.of(npiOrgNumber));
+    assertEquals("CAMPBELL CLINIC", npiOrgDisplay.get());
+  }
```

Comment:
Not sure how I feel about this. Ideally we shouldnt use the production file in our tests, but with the way the NPI thing is set up we can't test the functionality without doing so. For _testing purposes_ it'd be nice if instead of having a method that mentioned production, the method to set it up was just _createNpiOrgLookup([filestream])_ and you could pass in the method to use the test or production filestream (using the getFilestream method). This way we could use the testing file for these tests too.

(Maybe have one test to ensure the prod filestream exists, but i wouldnt test any data in it)